### PR TITLE
Update menu docs/examples and Add backward compatibility for menu icons

### PR DESCRIPTION
**Pull request by codetheorist**
AS-per [issue 10290](https://github.com/zurb/foundation-sites/issues/10290), update menu docs to specify adding '.icons' class as well as layout-specific class.

Also include examples of all four layout classes applied to the menu.

### CODE REVIEW

File: docs/pages/menu.md [hunk 1]
```diff
@@ -294,18 +294,59 @@ Menu items can have icons. Wrap the text of the item in a `<span>`, and then add
...
+  <li><a href="#"><span>One</span> <i class="fi-list"></i></a></li>
+  <li><a href="#"><span>Two</span> <i class="fi-list"></i></a></li>
+  <li><a href="#"><span>Three</span> <i class="fi-list"></i></a></li>
+  <li><a href="#"><span>Four</span> <i class="fi-list"></i></a></li>
+</ul>
+```
+### Icon Left
 <div class="docs-codepen-container">
-  <a class="codepen-logo-link" href="https://codepen.io/IamManchanda/pen/vmrYjN?editors=1000" target="_blank"><img src="{{root}}assets/img/logos/edit-in-browser.svg" class="" height="" width="" alt="edit on codepen button"></a>
+  <a class="codepen-logo-link" href="https://codepen.io/codetheorist/pen/XgqMMe?editors=1000" target="_blank"><img src="{{root}}assets/img/logos/edit-in-browser.svg" class="" height="" width="" alt="edit on codepen button"></a>
```
Comment (IamManchanda) [inline docs/pages/menu.md]: @codetheorist Can you update your codepen grid nested to it 
You can just fork => https://codepen.io/IamManchanda/pen/BZxWOj?editors=1000

**Comment by codetheorist** [comment]
@IamManchanda I have updated the link to the one provided by yourself.

**Comment by IamManchanda** [comment]
No we need to cut them in 4 diff codepen's ... wait let me ... I am digging in now ... 

Need to work on https://github.com/zurb/foundation-sites/issues/10290#issuecomment-312707724 too on this so let me work on it now :smile:

**Comment by IamManchanda** [comment]
@kball can we get a review on this from you ??

**Comment by kball** [comment]
Looks good. Nice work @codetheorist and @IamManchanda

**Pull request merged**